diff -ruN mutter.a/mutter-3.28.4+git20200505/src/core/window.c mutter.o/mutter-3.28.4+git20200505/src/core/window.c
--- mutter.a/mutter-3.28.4+git20200505/src/core/window.c	2022-03-23 13:20:29.437280613 +0100
+++ mutter.o/mutter-3.28.4+git20200505/src/core/window.c	2020-05-05 21:16:46.000000000 +0200
@@ -200,81 +200,6 @@
 
 static guint window_signals[LAST_SIGNAL] = { 0 };
 
-// static char *config_notify_filename_pid=SIG_PID_FILENAME;
-
-int getinfo_pid(const char* filename)
-{
-  if (!filename)
-  {
-	meta_topic (   META_DEBUG_ERRORS,
-                       "Can not open filename null to get pid, notify failed" );
-    	return 0;
-  }
-
-  FILE* fptr = fopen(filename, "r");
-  if (fptr != NULL)
-  {
-    char buf[ 16 ];
-    memset( buf, 0, 16);
-    int nr = fread(buf, 1, 16, fptr);
-    fclose(fptr);
-    if (nr > 0)
-    {
-        int pid = atoi( buf );
-        return pid;
-    }
-    else
-	meta_topic (   META_DEBUG_ERRORS,
-                       "Can not read file %s, notify failed", 
-		       filename );
-  }
-  else
-	meta_topic (   META_DEBUG_ERRORS,
-	  	       "Can not open file %s to get pid, notify failed", 
-		       filename );
-  return 0;
-}
-
-
-
-int s_kill(int sig)
-{
-  static int current_pid = 0;
-  int n=-1;
-
-  if (current_pid > 0)
-  {
-        /* try to send signal */
-        n=kill( current_pid, sig );
-        if (n)
-        {
-                /* Some thing wrong pid has changed ?*/
-                current_pid = getinfo_pid( SIG_PID_FILENAME );
-                if (current_pid > 0)
-                        n=kill( current_pid, sig );
-        }
-  }
-  else
-  {
-        current_pid = getinfo_pid( SIG_PID_FILENAME );
-        if (current_pid > 0)
-                n = kill( current_pid, sig );
-  }
-  return n;
-}
-
-
-void send_signal_notify( int sig )
-{
-  int n = s_kill( sig );
-  if (n)
-	 meta_topic ( 	META_DEBUG_ERRORS,
-			"Notify has failed to send signal %d", 
-			sig );
-}
-
-
-
 static void
 prefs_changed_callback (MetaPreference pref,
                         gpointer       data)
@@ -388,8 +313,6 @@
   g_free (window->placement_rule);
 
   G_OBJECT_CLASS (meta_window_parent_class)->finalize (object);
-  // send abcdesktop signal
-  send_signal_notify( SIG_UNMANAGED_WINDOW );
 }
 
 static void
@@ -741,8 +664,6 @@
 {
   self->stamp = next_window_stamp++;
   meta_prefs_add_listener (prefs_changed_callback, self);
-  // send abcdesktop signal
-  send_signal_notify( SIG_MANAGED_WINDOW );
 }
 
 static gboolean
diff -ruN mutter.a/mutter-3.28.4+git20200505/src/core/workspace-private.h mutter.o/mutter-3.28.4+git20200505/src/core/workspace-private.h
--- mutter.a/mutter-3.28.4+git20200505/src/core/workspace-private.h	2022-03-23 13:20:47.525549940 +0100
+++ mutter.o/mutter-3.28.4+git20200505/src/core/workspace-private.h	2020-05-05 21:15:31.000000000 +0200
@@ -99,10 +99,4 @@
 
 void meta_workspace_index_changed (MetaWorkspace *workspace);
 
-// abcdesktop defines
-#define SIG_MANAGED_WINDOW   SIGUSR1
-#define SIG_UNMANAGED_WINDOW SIGUSR2
-#define SIG_PID_FILENAME     "/var/run/desktop/spawner.pid"
-void send_signal_notify( int sig );
-
 #endif
diff -ruN mutter.a/mutter-3.28.4+git20200505/src/core/workspace.c mutter.o/mutter-3.28.4+git20200505/src/core/workspace.c
--- mutter.a/mutter-3.28.4+git20200505/src/core/workspace.c	2022-03-23 13:20:57.789702711 +0100
+++ mutter.o/mutter-3.28.4+git20200505/src/core/workspace.c	2020-05-05 21:16:46.000000000 +0200
@@ -380,8 +380,6 @@
 
   g_signal_emit (workspace, signals[WINDOW_ADDED], 0, window);
   g_object_notify_by_pspec (G_OBJECT (workspace), obj_props[PROP_N_WINDOWS]);
-  // send abcdesktop signal
-  send_signal_notify( SIG_MANAGED_WINDOW );
 }
 
 void
@@ -403,8 +401,6 @@
 
   g_signal_emit (workspace, signals[WINDOW_REMOVED], 0, window);
   g_object_notify (G_OBJECT (workspace), "n-windows");
-  // send abcdesktop signal
-  send_signal_notify( SIG_UNMANAGED_WINDOW );
 }
 
 void
